app/src/cose.h

```diff
@@ -207,6 +439,388 @@ namespace scitt::cose
+  {
+    UsefulBufC msg{cose_sign1.data(), cose_sign1.size()};
+
+    QCBORDecodeContext ctx;
+    QCBORDecode_Init(&ctx, msg, QCBOR_DECODE_MODE_NORMAL);
+
+    QCBORDecode_EnterArray(&ctx, nullptr);
+
+    QCBORItem item;
+
+    // skip body_protected
+    QCBORDecode_VGetNextConsume(&ctx, &item);
+
+    // skip unprotected header
+    QCBORDecode_VGetNextConsume(&ctx, &item);
+
+    // skip payload
+    QCBORDecode_VGetNextConsume(&ctx, &item);
+
+    // signature
+    QCBORDecode_GetNext(&ctx, &item);
+    auto signature = item.val.string;
+
+    QCBORDecode_ExitArray(&ctx);
+    auto error = QCBORDecode_Finish(&ctx);
+    if (error)
+    {
+      throw std::runtime_error("Failed to decode COSE_Sign1");
+    }
+
+    return qcbor_buf_to_vector(signature);
+  }
+
+  // Temporarily needed for notary_verify().
+  bool is_ecdsa_alg(int64_t cose_alg)
+  {
+    return cose_alg == COSE_ALGORITHM_ES256 ||
+      cose_alg == COSE_ALGORITHM_ES384 || cose_alg == COSE_ALGORITHM_ES512;
+  }
+
+  // Temporarily needed for notary_verify().
+  bool is_rsa_pss_alg(int64_t cose_alg)
+  {
+    return cose_alg == COSE_ALGORITHM_PS256 ||
+      cose_alg == COSE_ALGORITHM_PS384 || cose_alg == COSE_ALGORITHM_PS512;
+  }
+
+  // Temporarily needed for notary_verify().
+  crypto::MDType get_md_type(int64_t cose_alg)
+  {
+    switch (cose_alg)
+    {
+      case COSE_ALGORITHM_ES256:
+      case COSE_ALGORITHM_PS256:
+        return crypto::MDType::SHA256;
+      case COSE_ALGORITHM_ES384:
+      case COSE_ALGORITHM_PS384:
+        return crypto::MDType::SHA384;
+      case COSE_ALGORITHM_ES512:
+      case COSE_ALGORITHM_PS512:
+        return crypto::MDType::SHA512;
+      case COSE_ALGORITHM_EDDSA:
+        return crypto::MDType::NONE;
+      default:
+        throw std::runtime_error("Unsupported COSE algorithm");
+    }
+  }
+
+  // Temporarily needed for notary_verify().
+  const EVP_MD* get_openssl_md_type(crypto::MDType type)
+  {
+    switch (type)
+    {
+      case crypto::MDType::NONE:
+        return nullptr;
+      case crypto::MDType::SHA1:
+        return EVP_sha1();
+      case crypto::MDType::SHA256:
+        return EVP_sha256();
+      case crypto::MDType::SHA384:
+        return EVP_sha384();
+      case crypto::MDType::SHA512:
+        return EVP_sha512();
+      default:
+        throw std::runtime_error("Unsupported hash algorithm");
+    }
+    return nullptr;
+  }
+
+  // Temporarily needed for notary_verify().
+  static unsigned ecdsa_key_size(EVP_PKEY* key_evp)
+  {
+    int key_len_bits;
+    unsigned key_len_bytes;
+
+    key_len_bits = EVP_PKEY_bits(key_evp);
+
+    /* Calculation of size per RFC 8152 section 8.1 -- round up to
+     * number of bytes. */
+    key_len_bytes = (unsigned)key_len_bits / 8;
+    if (key_len_bits % 8)
+    {
+      key_len_bytes++;
+    }
+
+    return key_len_bytes;
+  }
+
+  // Temporarily needed for notary_verify().
+  std::vector<uint8_t> create_tbs(const std::vector<uint8_t>& cose_sign1)
```

Comment:
If you use `extract_sign1_fields` in the caller (which is needed to get the signature), you can change this
```suggestion
  std::vector<uint8_t> create_tbs(std::span<const uint8_t> protected, std::span<const uint8_t> payload)
```
and not need to do the decoding here.